ratis-server/src/test/java/org/apache/ratis/RaftBasicTests.java

```diff
@@ -337,6 +337,46 @@ public void testWithLoad() throws Exception {
     runWithNewCluster(NUM_SERVERS, cluster -> testWithLoad(10, 300, false, cluster, LOG));
   }
 
+  @Test
+  public void testPauseResume() throws Exception {
+    runWithNewCluster(NUM_SERVERS, this::runTestPauseResume);
+  }
+
+  void runTestPauseResume(CLUSTER cluster) throws InterruptedException, IOException {
+    // wait leader be elected.
+    RaftServerImpl leader = waitForLeader(cluster);
+    RaftPeerId leaderId = leader.getId();
+    List<RaftServerImpl> followers = cluster.getFollowers();
+    Assert.assertTrue(followers.size() >= 1);
+    RaftServerImpl follower = followers.get(0);
+
+    // keep sending messages to the leader.
+    SimpleMessage[] messages = SimpleMessage.create(100);
```

Comment:
Create two set of messages.  Send one set before pause and then send the other set after pause.  The paused follower should have all messages from the first set but none for the second set.